### PR TITLE
Telemetry Dashboard

### PR DESCRIPTION
**Pull request by aekeus**
User telemetry dashboard

```
  * Basic user authorization
  * Mongo data access
  * API
  * UI
  * DAU / Crashes
```

Please review authentication and access control: src/ui.js
@bbondy @yan

### CODE REVIEW

**Comment by bbondy** [comment]
@diracdeltas could you take?

**Comment by diracdeltas** [comment]
sure. @aekeus please cc me instead of @yan, who is probably confused why they keep coming up in our threads. :)

**Comment by aekeus** [comment]
Apologies :) @diracdeltas. will do.

File: src/ui.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+    const cache = server.cache({
+      segment: 'sessions',
+      expiresIn: 3 * 24 * 60 * 60 * 1000
+    })
+    server.app.cache = cache
+
+    // Auth strategy
+    server.auth.strategy('session', 'cookie', true, {
+      password: 'secret',
```
Comment (diracdeltas) [inline src/ui.js]: shouldn't this be in an env variable too? see https://github.com/brave/vault/issues/41

File: src/ui.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+      cookie: 'sid',
+      redirectTo: '/login',
+      isSecure: false,
```
Comment (diracdeltas) [inline src/ui.js]: why is this false? are we ever allowing this to be sent over non-HTTPS?

File: src/ui.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+// Setup authentication and user interface components
+exports.setup = (server) => {
+  // The ADMIN_PASSWORD environment variable must be set
+  if (!process.env.ADMIN_PASSWORD) {
+    throw new Error("ADMIN_PASSWORD not set")
+  }
+
+  // Register the server side templates
+  server.register(require('vision'), (err) => {
```
Comment (diracdeltas) [inline src/ui.js]: Is `err` supposed to be handled here?

File: src/ui.js [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+  // Auth library
+  server.register(require('hapi-auth-cookie'), (err) => {
+
+    const cache = server.cache({
+      segment: 'sessions',
+      expiresIn: 3 * 24 * 60 * 60 * 1000
+    })
+    server.app.cache = cache
+
+    // Auth strategy
```
Comment (diracdeltas) [inline src/ui.js]: can this use github auth like `vault` devops does?

**Comment by diracdeltas** [comment]
`npm start` complains `Error: MONGOLAB_URI must be set in environment`. what should i set it to?

**Comment by diracdeltas** [comment]
Probably not necessary yet, but authenticated actions that change server state should use CSRF tokens. https://github.com/hapijs/crumb

File: src/ui.js (hunk 1 above)
Comment (aekeus) [inline src/ui.js]: Absolutely, thanks.

File: src/ui.js (hunk 3 above)
Comment (aekeus) [inline src/ui.js]: Yes, will handle.

File: src/ui.js (hunk 4 above)
Comment (aekeus) [inline src/ui.js]: We could change this, not sure about timing though.

File: src/ui.js (hunk 2 above)
Comment (aekeus) [inline src/ui.js]: Probably a good idea to change. Will test.

**Comment by aekeus** [comment]
There currently are no authenticated update actions. But will add if we change.

**Pull request closed without merging**